tensornetwork/backends/jax/jitted_functions.py

```diff
@@ -518,20 +520,37 @@ def implicitly_restarted_arnoldi_method(
   return implicitly_restarted_arnoldi_method
 
 
-def gmres_wrapper(jax):
+def gmres_wrapper(jax: types.ModuleType) -> Dict:
   """
   Allows Jax (the module) to be passed in as an argument rather than imported,
-  since doing the latter breaks the build.
+  since doing the latter breaks the build. In addition, instantiates certain
+  of the enclosed functions as concrete objects within a Dict, allowing them to
+  be cached. This avoids spurious recompilations that would otherwise be
+  triggered by attempts to pass callables into Jitted functions.
+
+  The important function here is functions["gmres_m"], which implements
+  GMRES. The other functions are exposed only for testing.
+
   Args:
-    jax: The imported Jax module.
+  ----
+  jax: The imported Jax module.
+
   Returns:
-    gmres: A function performing gmres_m as described below.
+  -------
+  functions: A dictionary of functions:
+    functions["gmres_m"] = gmres_m
+    functions["gmres_residual"] = gmres_residual
+    functions["gmres_krylov"] = gmres_krylov
+    functions["_gs_step"] = _gs_step
+    functions["kth_arnoldi_step"] = kth_arnoldi_step
+    functions["givens_rotation"] = givens_rotation
```

Comment:
I don't like that this returns a dictionary. Perhaps use a NamedTuple instead?https://docs.python.org/3/library/collections.html#collections.namedtuple